src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -467,8 +470,13 @@ void dcd_init (uint8_t rhport)
 
   set_speed(rhport, TUD_OPT_HIGH_SPEED ? TUSB_SPEED_HIGH : TUSB_SPEED_FULL);
 
-  // Enable internal USB transceiver.
+  // Enable internal USB transceiver. Unconditional if using FS PHY on HS core.
+#if defined(TUD_OPT_SYNOPSYS_FS_PHY)
```

Comment:
same as above, but you could put the `OPT_MODE_HIGH_SPEED` into the if (condition)